waf.go

```diff
@@ -44,15 +45,15 @@ func NewWAF(config WAFConfig) (WAF, error) {
 		switch {
 		case r.rule != nil:
 			if err := waf.Rules.Add(r.rule); err != nil {
-				return nil, fmt.Errorf("invalid WAF config: %w", err)
```

Comment:
I think `%w` should be fine, even with tinygo, it doesn't use reflection AFAIU

Reply:
Why I like wrapping errors I think this is meaningless for now, are we doing `errors.Is` somewhere?

Reply:
This is a public API so it's unclear what a user may do with it. But there's no need to drop information anyways I think.

Reply:
Yeah and that is why I think it is too early to do wrapping without giving a thought on what errors are we returning (most of them are mainly anonymus) and when that time comes we can do wrapping, otherwise we are increasing the API surface and we can introduce breaking changes (if we replace the errors later) without us noticing and users noticing statically (imagine users expect ErrA to be wrapped and we change it to another error, we won't notice and users won't notice until something in runtime isn't behaving as expected)

Reply:
I guess that makes sense - I'd probably remove it from this PR and do it separately going through all the public APIs though